src/mypy_zope/plugin.py

```diff
@@ -479,6 +482,9 @@ def _analyze_zope_interface(
                 replacement = self._adjust_interface_function(api, cls.info, item)
             elif isinstance(item, OverloadedFuncDef):
                 replacement = self._adjust_interface_overload(api, cls.info, item)
+            elif isinstance(item, Decorator):
+                replacement = item
+                replacement.func = self._adjust_interface_function(api, cls.info, item.func)
```

Comment:
This seems to be causing error in the description. What this change is designed for?

The following patch to your branch makes all tests pass:

```diff
diff --git a/src/mypy_zope/plugin.py b/src/mypy_zope/plugin.py
index e9aeca7..02ab413 100644
--- a/src/mypy_zope/plugin.py
+++ b/src/mypy_zope/plugin.py
@@ -482,9 +482,6 @@ class ZopeInterfacePlugin(Plugin):
                 replacement = self._adjust_interface_function(api, cls.info, item)
             elif isinstance(item, OverloadedFuncDef):
                 replacement = self._adjust_interface_overload(api, cls.info, item)
-            elif isinstance(item, Decorator):
-                replacement = item
-                replacement.func = self._adjust_interface_function(api, cls.info, item.func)
             else:
                 continue
 
diff --git a/tests/samples/interface_meta.py b/tests/samples/interface_meta.py
index 2e04617..94d63fa 100644
--- a/tests/samples/interface_meta.py
+++ b/tests/samples/interface_meta.py
@@ -7,7 +7,7 @@ T = TypeVar('T', bound='zope.interface.Interface')
 class IBookmark(zope.interface.Interface):
     @staticmethod
     def create(url: str) -> 'IBookmark':
-        pass
+        return IBookmark(url)
 
 def createOb(iface: Type[T]) -> T:
     if zope.interface.interfaces.IInterface.providedBy(iface):
```

Reply:
Well, actually my patch is stupid, interface cannot have an implementation, disregard it...

Reply:
That `create` method didn't make too much sense anyways. I suggest this, unless there's something important about handling decorators:

```diff
diff --git a/src/mypy_zope/plugin.py b/src/mypy_zope/plugin.py
index e9aeca7..02ab413 100644
--- a/src/mypy_zope/plugin.py
+++ b/src/mypy_zope/plugin.py
@@ -482,9 +482,6 @@ class ZopeInterfacePlugin(Plugin):
                 replacement = self._adjust_interface_function(api, cls.info, item)
             elif isinstance(item, OverloadedFuncDef):
                 replacement = self._adjust_interface_overload(api, cls.info, item)
-            elif isinstance(item, Decorator):
-                replacement = item
-                replacement.func = self._adjust_interface_function(api, cls.info, item.func)
             else:
                 continue
 
diff --git a/tests/samples/interface_meta.py b/tests/samples/interface_meta.py
index 2e04617..35a7d42 100644
--- a/tests/samples/interface_meta.py
+++ b/tests/samples/interface_meta.py
@@ -5,8 +5,6 @@ T = TypeVar('T', bound='zope.interface.Interface')
 
 
 class IBookmark(zope.interface.Interface):
-    @staticmethod
-    def create(url: str) -> 'IBookmark':
         pass
 
 def createOb(iface: Type[T]) -> T:
@@ -20,6 +18,6 @@ def main(self) -> None:
 
 """
 <output>
-interface_meta.py:19: note: Revealed type is "__main__.IBookmark"
+interface_meta.py:17: note: Revealed type is "__main__.IBookmark"
 </output>
 """
```

Reply:
> What this change is designed for?

The `IBookmark` example that you cite. I thought the staticmethod was intended to be part of the `IBookmark` interface, so that classes which implement it would have to define their own `create` method.

If that's not the case then let's apply your patch!

Reply:
I don't even think interfaces can define static methods... At least it doesn't make sense to me today.

Reply:
> I suggest this, unless there's something important about handling decorators:

Ahh I see, the test doesn't even make use of `create`. Let's try the second patch.